hooks/useFetchProjects.ts

```diff
@@ -58,14 +59,45 @@ interface CachedProjects {
 const cacheKey = 'projectsData';
 const apiEndpoint = '/api/projects'; // The new API endpoint
 
+const extractTagsFromProjects = (projects: RepoItem[]) => {
+  return projects.reduce((acc: string[], project) => {
+    const tags = project.languages?.edges?.map(edge => edge.node.name);
+    tags.forEach(tag => {
+      if (!acc.includes(tag)) acc.push(tag);
+    });
+    return acc;
+  }, []);
+};
+
+const filterProjectsByTags = (projects: RepoItem[], tags: ProjectFilter[]) => {
+  if (!projects) return [];
+  if (!tags || tags.length === 0) return projects;
+
+  const projectsFiltered = projects.filter(project => {
+    const projectTags = project.languages?.edges?.map(edge => edge.node.name);
+    return projectTags?.some(tag =>
+      tags.some(t => t.name === tag && t.isActive)
+    );
+  });
+  console.log('🚀 ~ projectsFiltered ~ projectsFiltered:', {
+    projectsFiltered,
+  });
+  return projectsFiltered;
+};
+
 const useFetchProjects = () => {
   const [projects, setProjects] = useLocalStorage<CachedProjects>(cacheKey, {
     items: { items: [] },
     cacheTimestamp: Date.now().toString(),
   });
+  const [filteredProjects, setFilteredProjects] = useState<RepoItem[]>([]); // New state for the filtered projects
+  const [tags, setTags] = useState<string[]>([]);
```

Comment:
A bit confused, do you have an old state for the filtered projects? or this is the new state after your changes in the pr ?

Reply:
projects is the whole state fetched from external.
filteredProjects is the state displayed filtered from them.